python-package/lightgbm/callback.py

```diff
@@ -214,7 +215,23 @@ def _callback(env):
             _init(env)
         if not enabled[0]:
             return
+        if first_metric_only:
+            eval_metric = None
+            for metric_alias in ['metric', 'metrics', 'metric_types']:
+                if metric_alias in env.params.keys():
+                    if isinstance(env.params[metric_alias], (tuple, list)):
+                        eval_metric = env.params[metric_alias][0]
+                    else:
+                        eval_metric = env.params[metric_alias]
+                    break
         for i in range_(len(env.evaluation_result_list)):
+            metric_key = env.evaluation_result_list[i][1]
+            if metric_key.split(" ")[0] == "train":
+                continue  # train metric doesn't used on early stopping.
+            if first_metric_only:
+                if eval_metric is not None:
+                    if metric_key != "valid {}".format(eval_metric) and metric_key != eval_metric and eval_metric != "":
```

Comment:
Maybe merge this triple `if` cascade into one (or two, at least) with the help of `and`?